test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1242,13 +1254,13 @@ public synchronized void validateClusterFormed(String viaNode) {
     }
 
     @Override
-    public synchronized void afterTest() throws IOException {
+    public synchronized void afterTest() {
         wipePendingDataDirectories();
         randomlyResetClients(); /* reset all clients - each test gets its own client based on the Random instance created above. */
     }
 
     @Override
-    public void beforeIndexDeletion() throws Exception {
+    public synchronized void beforeIndexDeletion() throws Exception {
```

Comment:
I think synchronized here is now unnecessary?

Reply:
My motivation was to prevent any changes from happening concurrently between the individual assertions in this method, but I guess you could argue that that would simply indicate a bug with the tests.
=> I'll drop the synchronization here :)